gym_electric_motor/physical_systems/voltage_supplies.py

```diff
@@ -104,12 +104,86 @@ def reset(self):
         # Docstring of superclass
         # On reset the capacitor is loaded again
         self._solver.set_initial_value(np.array([self._u_0]))
-        self._u_sup = self._u_0
+        self._u_sup = [self._u_0]
         return self._u_sup
     
     def get_voltage(self, t,i_sup):
         # Docstring of superclass
         self._solver.set_f_params(self._u_0, i_sup, self._r, self._c)
-        self._u_sup = self._solver.integrate(t)[0]
+        self._u_sup = self._solver.integrate(t)
+        return self._u_sup
+    
+class AC1PhaseSupply(VoltageSupply):
+    """AC one phase voltage supply"""
+
+    def __init__(self, u_nominal=230, supply_parameter=None, **__):
+        """
+        Args:
+            u_nominal(float): Single phasic effective value of the voltage supply
+            supply_parameter(dict): Consists of frequency f in Hz, phase phi in range of [0,2*pi) 
+                                    and the flag fixed_phase which is set if the phase should be the same for each episode

```

Comment:
Thinking of the "fixed_phase" flag this might give only minor flexibility.
How about instead of that boolean flag provide a key "starting_angle" that, when given, is the fix starting angle, and if that key is not provided or comes with a value "None", we sample random starting angles at reset.